### PR TITLE
stop replacing double quotes in where strings

### PR DESCRIPTION
**Pull request by jgravois**
instead:
 * add an example and link to more SQL syntax help in the API reference
 * provide an error in the console explaining that doublequoted strings are a common reason for 400 errors
 * make existing tests syntactically correct

### CODE REVIEW

File: src/Tasks/Query.js [hunk 1]
```diff
@@ -97,14 +98,16 @@ EsriLeaflet.Tasks.Query = EsriLeaflet.Tasks.Task.extend({
...
-      }, context);
+      }, this);
 
     // otherwise convert it in the callback then pass it on
     } else {
       return this.request(function(error, response){
+        this._trapSQLerrors(error);
         callback.call(context, error, (response && EsriLeaflet.Util.responseToFeatureCollection(response)), response);
-      }, context);
+      }, this);
```
Comment (jgravois) [inline src/Tasks/Query.js]: context is `undefined` within this callback.  i needed to replaced with `this` so that i could call an internal method...

**Comment by jgravois** [comment]
closes #521

**Comment by patrickarlt** [comment]
@jgravois It looks like the tests get into some kind of infinite loop. Can you fix this an then I can merge this. https://travis-ci.org/Esri/esri-leaflet/builds/60904637

**Comment by jgravois** [comment]
ok. in order to fix the tests i needed to interrogate errors a little more deeply to confirm they were actually 400s from the server (and not readyState errors from within `Request`.  all is well now.

**Pull request merged**